containers/ddev-php-base/Dockerfile

```diff
@@ -32,6 +32,29 @@ RUN apt-get -qq install --no-install-recommends --no-install-suggests -y \
     wget
 #END base
 
+### ---------------------------ddev-xdebug-build--------------------------------------
+### Xdebug compile specific version because 3.3 is not fully reliable
+### See https://github.com/ddev/ddev/issues/6159
+### We don't need to recompile every Xdebug library https://xdebug.org/docs/compat (only PHP 8.0, 8.1, 8.2 can have Xdebug 3.2)
+### PECL does not allow you to install multiple versions of Xdebug, so there is `rm -f xdebug.reg`
+FROM base as ddev-xdebug-build
+SHELL ["/bin/bash", "-c"]
+RUN curl -sSLo /tmp/debsuryorg-archive-keyring.deb https://packages.sury.org/debsuryorg-archive-keyring.deb && \
+    dpkg -i /tmp/debsuryorg-archive-keyring.deb && rm -f /tmp/debsuryorg-archive-keyring.deb && \
+    echo "deb [signed-by=/usr/share/keyrings/deb.sury.org-php.gpg] https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list && apt-get update
+ARG XDEBUG_VERSION="3.2.2"
+ARG XDEBUG_BUILD_PACKAGES="build-essential php-pear php8.0-dev php8.1-dev php8.2-dev"
```

Comment:
Well, I used `ARG` because we don't want these variables to be inside the container:

```
$ docker run --rm -it ddev/ddev-php-base:v1.23.1 bash -c 'echo $XDEBUG_VERSION'
(nothing here)
```

See [Docker ARG vs ENV](https://vsupalov.com/docker-arg-vs-env/).

But when you declare it with `ENV`, it will be available inside, e.g.:

```
$ docker run --rm -it ddev/ddev-php-base:v1.23.1 bash -c 'echo $php83_arm64'
apcu bcmath bz2 curl cli common fpm gd imagick intl ldap mbstring memcached mysql opcache pgsql readline redis soap sqlite3 uploadprogress xdebug xhprof xml xmlrpc zip
```

I think it doesn't really matter here, since `ddev/ddev-php-base` isn't what users end up with, it's empty anyway:

```
$ docker run --rm -it ddev/ddev-webserver:v1.23.1 bash -c 'echo $php83_arm64'
(nothing here)
```

Reply:
That is awesome, thanks, always great to learn from you! I always use them only for when there's a --build-arg one way or another, but this is a great differentiator.